src/util/Path.cpp

```diff
@@ -272,29 +249,29 @@ GFile* Path::toGFile()
 /**
  * Get the parent path
  */
-Path Path::getParentPath()
+Path Path::getParentPath() const
 {
 	size_t separator = path.find_last_of("/\\");
 
 	if (separator == string::npos)
 	{
-		return "";
+		return Path{""};
 	}
 
-	return path.substr(0, separator);
+	return Path{path.substr(0, separator)};
 }
 
 /**
  * Convert an uri to a path, if the uri does not start with file:// an empty Path is returned
  */
-Path Path::fromUri(string uri)
+Path Path::fromUri(const string& uri)
```

Comment:
Do not change this, else it's not possible to call this from anything else than a std::string, keep it consistent over the whole API of the class.

Reply:
Actually that's not correct. A `Type const&` binds to everything also to temporaries (as `auto&&` or `Type&&`) since string has no explicit Constructors (it has a converting) constructor it will work without a problem.
It is actually more efficient.

Reply:
Checked an compiling with g++ 7.4

Not sure when this was changed: But this was definite not working with older versions, the main reason why I added nearly nowhere const to the parameter...

Reply:
Then the additional const char* overload methods also should not be needed, right?
I meand operator/= etc.

Reply:
I'll do a rework to that class, seems like i can clean up more :D
And it should be supported since the first compiler supporting c++11. So somewhere around 2009.

Reply:
Ok the functions with `char const*` aren't needed, but if I remove them, everytime someone passes a cstring it will be converted to a std::string first, which is not necessary.

Reply:
Ok